### ISSUE TITLE
Feature: a way to disable the web preview or just display a screenshot instead

### ISSUE THREAD

**Issue by adrian7** [docker/kitematic#1455]
I'm using Kitematic 0.9.3 on Mac OS X.

I have a project (my bad) which does a lot of javascript. Unfortunately this causes Kitematic UI to freeze until the website loads in web preview. 
I would like to be able to disable web preview or have that component display a screenshot instead of loading the page and executing all the scripts.

Regards,
Adrian.

**Comment by FrenchBen** [comment]
Thanks for the info - Would  you be able to write-up how you see it working?
This would help anyone willing to work on this feature.

**Comment by adrian7** [comment]
I think ideally in the View menu, have an item 

Web preview 
 - Browser
 - Screenshot only
 - Disable

**Comment by alexandrev** [comment]
@FrenchBen, what do you think about the approach? 3 options? Only a check to disable it? If you want I could go for it if you think this is important..

**Comment by FrenchBen** [comment]
@alexandrev I don't think this is important, as this is the first use case - I would like the OP to take a stab at it, if he thinks this is something important to him.
I'm sure it's a nice feature, but there's a bit more to it due to the info for that container no longer being saved in the container data but rather in somewhere in kitematic.

**Comment by ericmann** [comment]
Chiming in here because Kitematic's built-in browser view gave me a bit of a headache today. It's not that I'm running a container that prints a lot of JS, it's that my container is attempting to parse specific traffic sent to a TCP port it monitors. Normally, everything's fine. But if I alt-tab to the Kitematic window, my app will crash because of the flood of _web_ traffic it gets from the browser attempting to hit the open port directly.

Being able to turn this off would be a huge win for me.

**Comment by FrenchBen** [comment]
@ericmann As explained above, I'm waiting for someone to step up and provide details on how this feature will be implemented.
Currently we're focused on bug fixes rather features like this. Any help is appreciated however.

**Comment by bostonaholic** [comment]
I'd like to reopen this feature exploration with PR #2016

**Comment by FrenchBen** [comment]
@bostonaholic Since you're actually moving on the PR, I'm re-opening. Thanks for carrying this.

**Comment by jpduckwo** [comment]
my kitematic crashes every time I open it and click on my nginix container which hosts an angular js app. I would also like to disable the web preview which I think is causing the crash.

**Comment by scytacki** [comment]
A work around that might work in some cases is to change which port your nginix container is using internally. Kitematic uses this set of "webPorts" to decide if it should open the web preview: https://github.com/docker/kitematic/blob/e40ba41f1b2d623636241162df0b44178a685af6/src/utils/Util.js#L215

**Comment by npendery** [comment]
@scytacki how do I get to that file? cant seem to find it

**Comment by scytacki** [comment]
@npendery the snippet from file posted above is from the Kitematic source. I don't know enough about Kitematic to know if you could successfully change it locally.  I was suggesting that as a work around developers could change their own Docker and app configuration. The change is to the internal port of the container. If this internal port is not in the list above, then Kitematic will not open the web preview.

For reference, this issue was causing one of my containers to crash because it didn't expect an http GET request on its port.
